### PR TITLE
store delegation certificates to database

### PR DESCRIPTION
**Pull request by crchemist-ip**

### CODE REVIEW

File: src/blockchain/shelley/certificate.js [hunk 1]
```diff
@@ -23,19 +27,10 @@ export type PoolUpdate = {
...
-  pool_id: ?PoolId,
-  account: string,
-}
-
-export type OwnerStakeDelegation = {
-  type: 'OwnerStakeDelegation',
-  // null here signifies NonDelegated, ie un-delegate
-  pool_id: ?PoolId,
-  // this is not in the certificate, but was inferred from the tx
+export type StakeDelegationType = {
```
Comment (rooooooooob) [inline src/blockchain/shelley/certificate.js]: Why was this changed? If we're going to be combining `OwnerStakeDelegation` and `StakeDelegation` we can't use the `type` field as we're already using this as the tag in the `CertificateType` tagged enum.

File: src/blockchain/shelley/utils.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
-// flow
+// @flow
 
-import { TxInputType } from '../common'
+import { CERT_TYPE } from './certificate'
 
-const fragmentToObj = (fragment: any, extraData: {}): TxType => {
+const fragmentToObj = (fragment: any, extraData: {}) => {
```
Comment (rooooooooob) [inline src/blockchain/shelley/utils.js]: Why is the return type removed here?

File: src/blockchain/shelley/certificate.js [hunk 3]
```diff
@@ -23,19 +23,10 @@ export type PoolUpdate = {
...
+  type: number,
```
Comment (vantuz-subhuman) [inline src/blockchain/shelley/certificate.js]:
```suggestion
  type: 'StakeDelegation' | 'OwnerStakeDelegation',
```

Comment (vantuz-subhuman) [inline src/blockchain/shelley/certificate.js]: @crchemist-ip, I agree with combining the types together, but after some discussion we are agreeing with @rooooooooob that this `type` field is supposed to be a string, to be consistent with all types used in the `CertificateType` union.

So here we should change this line to:
```
type: 'StakeDelegation',
```

But then also let's add flag like `isOwnerStake` - and the parser can set that flag instead of creating separate type like before.

File: src/blockchain/shelley/utils.js [hunk 4]
```diff
@@ -104,8 +106,8 @@ const fragmentToObj = (fragment: any, extraData: {}): TxType => {
         const deleg = cert.get_stake_delegation()
         const poolId = deleg.delegation_type().get_full()
         common.certificate = {
-          type: 'StakeDelegation',
-          // TODO: possibly handle Ratio types
+          type: wasm.CertificateType.StakeDelegation,
+          // TODO: handle DelegationType parsing
```
Comment (vantuz-subhuman) [inline src/blockchain/shelley/utils.js]: according to this comment: https://github.com/Emurgo/tangata-manu/pull/99/files#r350673514

Here type should also be changed back to `type: 'StakeDelegation'`

File: src/blockchain/shelley/utils.js [hunk 5]
```diff
@@ -139,7 +141,8 @@ const fragmentToObj = (fragment: any, extraData: {}): TxType => {
         break
```
Comment (vantuz-subhuman) [inline src/blockchain/shelley/utils.js]: If we have combined the delegation certificate types together: https://github.com/Emurgo/tangata-manu/pull/99/files#r350673514

Here the handling for the `wasm.CertificateType.OwnerStakeDelegation` should also be changed. The `type` in the resulting object should be `StakeDelegation` and additional field `isOwnerStake: true` added

File: src/blockchain/shelley/utils.js [hunk 6]
```diff
@@ -151,25 +154,9 @@ const fragmentToObj = (fragment: any, extraData: {}): TxType => {
...
-  //     const [address, value] = out
-  //     return { address: bs58.encode(cbor.encode(address)), value }
-  //   }),
-  //   txBody,
-  //   ...extraData,
-  // }
 }
 
-const rawTxToObj = (tx: Array<any>): TxType => {
+const rawTxToObj = (tx: Array<any>) => {
```
Comment (vantuz-subhuman) [inline src/blockchain/shelley/utils.js]: Why the return type removed as well? 🤔

File: src/blockchain/shelley/utils.js (hunk 2 above)
Comment (crchemist-ip) [reply src/blockchain/shelley/utils.js]: There is no such time in codebase.I suggest there should be something like `ShelleyTxType`

Comment (vantuz-subhuman) [reply src/blockchain/shelley/utils.js]: ah, ok makes sense for now
